packages/next/client/head-manager.ts

```diff
@@ -34,46 +35,19 @@ function reactElementToDOM({ type, props }: JSX.Element): HTMLElement {
 
 function updateElements(type: string, components: JSX.Element[]) {
   const headEl = document.getElementsByTagName('head')[0]
-  const headCountEl: HTMLMetaElement = headEl.querySelector(
-    'meta[name=next-head-count]'
-  ) as HTMLMetaElement
-  if (process.env.NODE_ENV !== 'production') {
-    if (!headCountEl) {
-      console.error(
-        'Warning: next-head-count is missing. https://err.sh/next.js/next-head-count-missing'
-      )
-      return
-    }
-  }
-
-  const headCount = Number(headCountEl.content)
-  const oldTags: Element[] = []
+  const headCountEls = headEl.querySelectorAll('[data-next-head=true]')
 
-  for (
-    let i = 0, j = headCountEl.previousElementSibling;
-    i < headCount;
-    i++, j = j!.previousElementSibling
-  ) {
-    if (j!.tagName.toLowerCase() === type) {
-      oldTags.push(j!)
-    }
-  }
-  const newTags = (components.map(reactElementToDOM) as HTMLElement[]).filter(
-    (newTag) => {
-      for (let k = 0, len = oldTags.length; k < len; k++) {
-        const oldTag = oldTags[k]
-        if (oldTag.isEqualNode(newTag)) {
-          oldTags.splice(k, 1)
-          return false
-        }
-      }
-      return true
+  headCountEls.forEach((el) => {
+    if (el!.tagName.toLowerCase() === type) {
+      headEl.removeChild(el)
     }
-  )
+  })
 
-  oldTags.forEach((t) => t.parentNode!.removeChild(t))
-  newTags.forEach((t) => headEl.insertBefore(t, headCountEl))
-  headCountEl.content = (headCount - oldTags.length + newTags.length).toString()
+  components.forEach((t) => {
+    // TODO not getting react components here
+    console.log(t)
+    headEl.appendChild(reactElementToDOM(t))
```

Comment:
@devknoll still unsure how to handle the CSR bit... but I got the SSR stuff rendered :)

Reply:
> TODO not getting react components here

derp, because of 
```
['meta', 'base', 'link', 'style', 'script'].forEach((type) => {
          updateElements(type, tags[type] || [])
        })
```